### PR TITLE
Destroy now cleans up all goroutines opened by the SDK

### PR DESCRIPTION
**Pull request by crimsonred**

### CODE REVIEW

File: listener_manager.go [hunk 1]
```diff
@@ -62,11 +62,12 @@ func (m *ListenerManager) announceStatus(status *PNStatus) {
 	go func() {
 		m.RLock()
 		m.pubnub.Config.Log.Println("announceStatus lock")
+	B:
 		for l := range m.listeners {
 			select {
 			case <-m.exitListener:
 				m.pubnub.Config.Log.Println("announceStatus exitListener")
-				break
+				break B
```
Comment (davidnub) [inline listener_manager.go]: Can we use a different label here that is more clear?

File: request_workers.go [hunk 2]
```diff
@@ -29,42 +29,65 @@ type RequestWorkers struct {
...
 		JobChannel: make(chan *JobQItem),
+		ctx:        ctx,
 		id:         id,
 	}
 }
 
+// Process runs a goroutine for the worker
 func (pw Worker) Process(pubnub *PubNub) {
 	go func() {
+	B:
```
Comment (davidnub) [inline request_workers.go]: Can we use a more descriptive label here?

File: request_workers.go [hunk 3]
```diff
@@ -29,42 +29,65 @@ type RequestWorkers struct {
...
-			res, err := job.Client.Do(job.Req)
-			jqr := &JobQResponse{
-				Error: err,
-				Resp:  res,
+			select {
+			case pw.Workers <- pw.JobChannel:
+				job := <-pw.JobChannel
+
+				if job != nil {
+
```
Comment (davidnub) [inline request_workers.go]: Nitpick. Why the extra whitespace?

File: telemetry_manager.go [hunk 4]
```diff
@@ -118,27 +117,17 @@ func (m *TelemetryManager) startCleanUpTimer() {
 			cleanUpInterval*cleanUpIntervalMultiplier) * time.Millisecond)
 
 	go func() {
+	B:
```
Comment (davidnub) [inline telemetry_manager.go]: Can we use a more descriptive label here?

File: request_workers.go (hunk 3 above)
Comment (crimsonred) [reply request_workers.go]: Removed

File: listener_manager.go (hunk 1 above)
Comment (crimsonred) [reply listener_manager.go]: Modified to AnnounceStatusLabel / AnnounceMessageLabel

File: request_workers.go (hunk 2 above)
Comment (crimsonred) [reply request_workers.go]: Modified to ProcessLabel

File: telemetry_manager.go (hunk 4 above)
Comment (crimsonred) [reply telemetry_manager.go]: Modified to CleanUpTimerLabel

**Pull request merged**